server/db/driver/pg/orders.go

```diff
@@ -189,6 +188,39 @@ func (a *Archiver) NewEpochOrder(ord order.Order, epochIdx, epochDur int64) erro
 	return a.storeOrder(ord, epochIdx, epochDur, orderStatusEpoch)
 }
 
+// BookOrders retrieves all booked orders (with order status booked) for the
+// specified market. This will be used to repopulate a market's book on
+// construction of the market.
+func (a *Archiver) BookOrders(base, quote uint32) ([]*order.LimitOrder, error) {
+	marketSchema, err := a.marketSchema(base, quote)
+	if err != nil {
+		return nil, err
+	}
+
+	// All booked orders are active.
+	tableName := fullOrderTableName(a.dbName, marketSchema, true) // active (true)
+
+	// no query timeout here, only explicit cancellation
+	ords, err := ordersByStatusFromTable(a.ctx, a.db, tableName, base, quote, orderStatusBooked)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify loaded orders are limits, and cast to *LimitOrder.
+	limits := make([]*order.LimitOrder, len(ords))
+	for _, ord := range ords {
+		lo, ok := ord.(*order.LimitOrder)
```

Comment:
`ordersByStatusFromTable` can be used for any order status and thus could include market orders.  But it's an error if a booked order is also a market order.  Should never happen, but I'd like to know if it does.